cogs/subjects.py

```
@@ -0,0 +1,272 @@
+
+    @subject.command(pass_context=True,
+                     name="show")
+    async def subject_show(self, ctx: Context):
+        """
+        Shows all possible opt-in and opt-out subjects of the user
+
+        Args:
+            ctx: The command context provided by the discord.py wrapper.
+        """
+        member: Union[Member, User] = ctx.author
+        roles: list[Role] = member.roles
+
+        subjects: list[Role] = await self.get_possible_subjects(ctx, member, roles)
+
+        if not subjects:
+            return
+
+        embed = Embed(title="Subjects",
+                      description="You can opt-in/out of following subjects:")
+        subjects_text: str = str([subject.name for subject in subjects if subject in roles])
+        embed.add_field(name="Opt-out Subjects", value=subjects_text[1:-1].replace("'", "`").replace(",", "\n"),
+                        inline=False)
+        subjects_text: str = str([subject.name for subject in subjects if subject not in roles])
+        embed.add_field(name="Opt-in Subjects", value=subjects_text[1:-1].replace("'", "`").replace(",", "\n"),
+                        inline=False)
+
+        await ctx.reply(embed=embed)
+
+    @subject.command(pass_context=True,
+                     name="add")
+    async def subject_add(self, ctx: Context, subject: str):
+        """
+        opts-in a user into the specified subject
+
+        Args:
+            ctx: The command context provided by the discord.py wrapper.
+
+            subject: The Subject to opt into
+        """
+        member: Union[Member, User] = ctx.author
+        roles: list[Role] = member.roles
+
+        subjects: list[Role] = await self.get_possible_subjects(ctx, member, roles)
+
+        if not subjects:
+            return
+
+        if subject not in [subject.name for subject in subjects]:
+            embed = Embed(title="No permission",
```

This should throw an error so that it can be treated as an error.